src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -739,7 +739,9 @@ static void ensure_join(JavaThread* thread) {
   // Thread is exiting. So set thread_status field in  java.lang.Thread class to TERMINATED.
   java_lang_Thread::set_thread_status(threadObj(), JavaThreadStatus::TERMINATED);
   // Clear the native thread instance - this makes isAlive return false and allows the join()
-  // to complete once we've done the notify_all below
+  // to complete once we've done the notify_all below. Needs a release() to obey Java Memory Model
+  // requirements.
```

Comment:
This might be more readable:
```suggestion
  // to complete once we've done the notify_all below.
  // Needs a release() to obey Java Memory Model requirements.
```

Reply:
Multi-line comments are typically written in normal paragraph style.